packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -135,157 +195,199 @@ describe("Presence", () => {
+						};
+
+						// Join response signal broadcasted from initial joining attendee (responding to the second joining attendee message sent above)
+						const reponseSignalInfo = {
+							clientSessionId: attendeeSessionId,
+							clientConnectionId: initialAttendeeConnectionId,
+							fixedTime: clock.now - 30,
+							averageLatency: 20,
+							// Include the prior client to session id mapping info for the second joining attendee
+							priorClientToSessionId: {
+								"client9": {
+									rev: 0,
+									timestamp: clock.now - 40,
+									value: "sessionId-9",
+								},
+							},
+						};
+
+						// ACT - simulate join messages from clients
+						// Order matters here: 1st client joins -> 2nd client joins -> 1st client responds to 2nd client joining
+						const joinedAttendees: ISessionClient[] = simulateAttendeeJoin([
+							initialAttendeeSignalInfo,
+							secondJoinSignalInfo,
+							reponseSignalInfo,
+						]);
+
+						// VERIFY
+						assert(
+							joinedAttendees.length === 2 &&
+								joinedAttendees[0] !== undefined &&
+								joinedAttendees[1] !== undefined,
+							"Expected exactly two attendees to be announced",
+						);
+
+						verifyAttendee(joinedAttendees[0], initialAttendeeConnectionId);
+						verifyAttendee(joinedAttendees[1], secondJoinSignalInfo.clientConnectionId);
 					});
 				});
 
 				describe("that is already known", () => {
 					let knownAttendee: ISessionClient | undefined;
 
 					beforeEach(() => {
-						const removeListener = presence.events.on("attendeeJoined", (attendee) => {
-							knownAttendee = attendee;
-						});
-
-						// Setup - simulate join message from client
-						presence.processSignal("", initialAttendeeSignal, false);
-						assert(knownAttendee !== undefined, "No attendee was announced in setup");
-						removeListener();
+						// Setup known attendee
+						const joinedAttendees = simulateAttendeeJoin([initialAttendeeSignalInfo]);
+						assert(
+							joinedAttendees.length === 1,
+							"Expected exactly one attendee to be announced",
+						);
+						knownAttendee = joinedAttendees[0];
 					});
 
 					it('is NOT announced when "rejoined" with same connection (duplicate signal)', () => {
-						afterCleanUp.push(
-							presence.events.on("attendeeJoined", (attendee) => {
-								assert.fail(
-									"Attendee should not be announced when rejoining with same connection",
-								);
-							}),
-						);
-
 						clock.tick(10);
-
-						// Act & Verify - simulate duplicate join message from client
-						presence.processSignal("", initialAttendeeSignal, false);
+						// Act - simulate duplicate join message from client
+						const joinedAttendees = simulateAttendeeJoin([initialAttendeeSignalInfo]);
+						// Verify - no attendee should be announced for duplicate join message
+						assert(
+							joinedAttendees.length === 0,
+							"No attendee should be announced for duplicate join message",
+						);
```

Comment:
The advantage of the prior version is that if the event were fired, the failure would happen exactly when the unexpected event fired instead of sometime after. Prior is much easier to debug. Should be able to keep the prior verification and use the new helper (not need to check the result).